File: Easy/0027-Remove-Element/Solution.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    if (nums.length === 0) return 0;

  let k = 0;
  let i = 0;

  while (i < nums.length) {
    if (nums[i] !== val) {
      nums[k] = nums[i];
      k++;
    }
    i++;
  }

  return k;
};

File: Easy/0027-Remove-Element/Solution.rb
# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
    return 0 if nums.length == 0
    k = 0
    i = 0
    while i < nums.length
        if nums[i] != val
            nums[k] = nums[i]
            k += 1
        end
        i += 1
    end
    k
end